opent2t.p.outlet/com.wink.binaryswitch/js/thingTranslator.js

```diff
@@ -0,0 +1,166 @@
+/* jshint esversion: 6 */
+/* jshint node: true */
+'use strict';
+
+// This code uses ES2015 syntax that requires at least Node.js v4.
+// For Node.js ES2015 support details, reference http://node.green/
+
+function validateArgumentType(arg, argName, expectedType) {
+    if (typeof arg === 'undefined') {
+        throw new Error('Missing argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + '.');
+    } else if (typeof arg !== expectedType) {
+        throw new Error('Invalid argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + ', got: ' + (typeof arg));
+    }
+}
+
+// Wink does not always populate every desired_state property, but last_reading doesn't necessarily
+// update as soon as we send our PUT request. Instead of relying just on one state or the other,
+// we use this StateReader class to read from desired_state if it is there, and fall back to last_reading
+// if it is not.
+class StateReader {
+    constructor(desired_state, last_reading) {
+        this.desired_state = desired_state;
+        this.last_reading = last_reading;
+    }
+
+    get(state) {
+        if (this.desired_state[state] !== undefined) {
+            return this.desired_state[state];
+        }
+        else {
+            return this.last_reading[state];
+        }
+    }
+}
+
+// Helper method to convert the provider schema to the platform schema.
+function providerSchemaToPlatformSchema(providerSchema, expand) {
+    var stateReader = new StateReader(providerSchema.desired_state, providerSchema.last_reading);
+
+    var powered = stateReader.get('powered');
+
+    var power = {
+        href: '/power',
+        rt: ['oic.r.switch.binary'],
+        if: ['oic.if.a', 'oic.if.baseline']
+    };
+    
+    if (expand) {
+        power.id = 'power';
+        power.value = powered;
+    }
+
+    return {
+        opent2t: {
+            schema: 'opent2t.p.outlet',
+            translator: 'opent2t-translator-com-wink-binaryswitch',
+            controlId: deviceId
+        },
+        pi: providerSchema['uuid'],
+        mnmn: providerSchema['device_manufacturer'],
+        mnmo: providerSchema['manufacturer_device_model'],
+        n: providerSchema['name'],
+        rt: ['opent2t.p.outlet'],
+        entities: [
+            {
+                rt: ['oic.d.smartplug'],
+                di: lightDeviceDi,
+                resources: [
+                    power
+                ]
+            }
+        ]
+    };
+}
+
+// Helper method to convert the translator schema to the device schema.
+function resourceSchemaToProviderSchema(resourceId, resourceSchema) {
+
+    // build the object with desired state
+    var result = { 'desired_state': {} };
+    var desired_state = result.desired_state;
+
+    if ('power' === resourceId) {
+        desired_state['powered'] = resourceSchema.value;
+    }
+
+    return result;
+}
+
+function findResource(schema, di, resourceId) {
```

Comment:
Agreed it would be nice at some point. Not sure where it would go right now though.